src/coreclr/jit/earlyprop.cpp

```diff
@@ -762,8 +762,7 @@ bool Compiler::optCanMoveNullCheckPastTree(GenTree* tree,
             else if (isInsideTry)
             {
                 // Inside try we allow only stores to locals not live in handlers.
-                // lvVolatileHint is set to true on variables that are line in handlers.
-                result = tree->OperIs(GT_STORE_LCL_VAR) && !lvaTable[tree->AsLclVar()->GetLclNum()].lvVolatileHint;
+                result = tree->OperIs(GT_STORE_LCL_VAR) && !lvaTable[tree->AsLclVar()->GetLclNum()].lvLiveInOutOfHndlr;
```

Comment:
Looks like we didn't use this originally because it was debug-only back then:
https://github.com/dotnet/runtime/pull/1735#discussion_r367268487

However, today we do have it in all flavors and it gets computed right before early prop.